fix transient dialog placement
dialogs can be bigger than the mainwindow and the unsigned dimensions
then overflow in the subtraction (the window would still be moved into
screen bounds but appear on ugly 0,0)

diff --git a/src/Window.cc b/src/Window.cc
index 4410ffe..b93a164 100644
--- a/src/Window.cc
+++ b/src/Window.cc
@@ -490,6 +490,9 @@ void FluxboxWindow::init() {
         layerItem().setLayer(twin->layerItem().getLayer());
         m_state.layernum = twin->layerNum();
         m_workspace_number = twin->workspaceNumber();
+        const int x = twin->frame().x() + int(twin->frame().width() - frame().width())/2;
+        const int y = twin->frame().y() + int(twin->frame().height() - frame().height())/2;
+        frame().move(x, y);
     } else // if no parent then set default layer
         moveToLayer(m_state.layernum, m_state.layernum != ::ResourceLayer::NORMAL);
 
